Sources/OpenAPIServiceModel/CreateOpenAPIServiceModel.swift

```diff
@@ -0,0 +1,382 @@
+                
+                let inputDescription =
+                        OperationInputDescription(defaultInputLocation: .body)
+                
+                var operationDescription = OperationDescription(
+                    inputDescription: inputDescription,
+                    outputDescription: OperationOutputDescription())
+                operationDescription.httpUrl = path.rawValue
+                operationDescription.httpVerb = type.rawValue.uppercased()
+                
+                parseOperation(description: &operationDescription,
+                               operationName: operationName,
+                               model: &model, operation: operation,
+                               modelOverride: modelOverride, document: definition)
+                
+                model.operationDescriptions[operationName] = operationDescription
+            }
+        }
+        
+        return model
+    }
+    
+    static func parseOperation(description: inout OperationDescription,
+                               operationName: String,
+                               model: inout OpenAPIServiceModel,
+                               operation: OpenAPI.Operation,
+                               modelOverride: ModelOverride?,
+                               document: OpenAPI.Document) {
+        let (members, bodyStructureName) = getOperationMembersAndBodyStructureName(
+            operation: operation,
+                            operationName: operationName,
+                            model: &model,
+                            modelOverride: modelOverride, document: document)
+        
+        setOperationInput(bodyStructureName: bodyStructureName, operationInputMembers: members,
+                          description: &description, model: &model, operationName: operationName)
+        
+        setOperationOutput(operation: operation, operationName: operationName, model: &model,
+                           modelOverride: modelOverride, description: &description, document: document)
+    }
+    
+    static func getOperationMembersAndBodyStructureName(
+            operation: OpenAPI.Operation,
+            operationName: String,
+            model: inout OpenAPIServiceModel,
+            modelOverride: ModelOverride?, document: OpenAPI.Document) -> (members: OperationInputMembers, bodyStructureName: String?) {
+        var members = OperationInputMembers()
+        var bodyStructureName: String?
+        
+        if let requestBody = operation.requestBody?.b {
+            getBodyOperationMembers(requestBody, bodyStructureName: &bodyStructureName,
+                                    operationName: operationName, model: &model, modelOverride: modelOverride, document: document)
+        } else {
+            fatalError("no request body")
+        }
+        
+        for (index, parameter) in operation.parameters.enumerated() {
+            switch parameter {
+            case .b(let value):
```

Comment:
nit: maybe rename `value` to `parameterValue` or something so it a bit more clear what the variable is.

Reply:
done